### ISSUE TITLE
Unable to get docker updates for Ubuntu 14.04 LTS Trusty Tahr

### ISSUE THREAD

**Issue by Roan89** [docker/docker#16941]
I was trying to update(apt-get update) my system an the following errors were notified

    Err https://apt.dockerproject.org ubuntu-trusty/main amd64 Packages            
      Received HTTP code 403 from proxy after CONNECT
    Err https://apt.dockerproject.org ubuntu-trusty/main i386 Packages             
      Received HTTP code 403 from proxy after CONNECT
    .
    .
    .
    W: Failed to fetch https://apt.dockerproject.org/repo/dists/ubuntu-trusty/main/binary-amd64/Packages  Received HTTP code 403 from proxy after CONNECT

    W: Failed to fetch https://apt.dockerproject.org/repo/dists/ubuntu-trusty/main/binary-i386/Packages Received HTTP code 403 from proxy after CONNECT

    E: Some index files failed to download. They have been ignored, or old ones used instead.

Below is my OS version and docker details

     Docker Version : Docker version 1.8.2, build 0a8c2e3
    Docker info : 
    Containers: 25
    Images: 12
    Storage Driver: aufs
    Root Dir: /var/lib/docker/aufs
    Backing Filesystem: extfs
    Dirs: 62
    Dirperm1 Supported: false
    Execution Driver: native-0.2
    Logging Driver: json-file
    Kernel Version: 3.13.0-65-generic
    Operating System: Ubuntu 14.04.3 LTS
    CPUs: 8
    Total Memory: 7.687 GiB
    Name: workspace
    ID: AL47:NZ7R:F2LN:7SZQ:U7BQ:2E5C:P4MS:ZWG6:T5H6:NY5Z:ATD2:JIT3
    WARNING: No swap limit support

    uname -a : Linux workspace 3.13.0-65-generic #106-Ubuntu SMP Fri Oct 2 22:08:27 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
   

The website I followed to install docker : https://docs.docker.com/installation/ubuntulinux/ 

Why does this happen?
Can this issue affect other programs that use docker ?

**Comment by Roan89** [comment]
changing http to https in the sources.list.d/docker.list to fixed the issue

**Comment by lmfmkos** [comment]
Did not helped :(

**Comment by byk0t** [comment]
Use http instead of https in */etc/apt/sources.list.d/docker.list*. It helped me.

**Comment by VimleshS** [comment]
changing to http worked for me.

**Comment by FelikZ** [comment]
https -> http helps, but man in the middle is happy now

**Comment by sandeep-sarpe** [comment]
changing to http worked for me too.

**Comment by olojiang** [comment]
changed from https => http works for me too.

**Comment by ewhitmore** [comment]
http did not work for me:

`Fetched 4,998 B in 14s (336 B/s)
W: Failed to fetch http://apt.dockerproject.org/repo/dists/ubuntu-trusty/main/binary-amd64/Packages  Hash Sum mismatch

E: Some index files failed to download. They have been ignored, or old ones used instead.
`

**Comment by zloylos** [comment]
+1 
```
Fetched 32.9 MB in 21s (1536 kB/s)
W: Failed to fetch http://apt.dockerproject.org/repo/dists/ubuntu-trusty/main/binary-amd64/Packages  Hash Sum mismatch

E: Some index files failed to download. They have been ignored, or old ones used instead.
```

**Comment by thaJeztah** [comment]
We're currently having an issue with the APT repository (see https://github.com/docker/docker/issues/23203). I raised this issue internally, and we're trying to get this fixed ASAP

As a temporary workaround, you can install docker 1.11.2-rc1 from the "test" repository; 1.11.2-rc1 is almost the same as the current release, apart from these three changes;
https://github.com/docker/docker/pull/23164, https://github.com/docker/docker/pull/23169, and https://github.com/docker/docker/pull/23176

Those changes should not make a functional difference (and the last change only affects some corner-cases)

You can install the RC, either by changing the "main" to "test" repository for APT, or using the install script;

    curl -fsSL https://test.docker.com | sh

Remember to switch back to the regular (main) repository after the issue is resolved

**Comment by sudheeshms** [comment]
https -> **http**

**Comment by shrmrf** [comment]
Hello Guys, has this issue resurfaced today? Fix onda way?

I'm testing on a fresh ubuntu VM.

```bash
$ lsb_release -a
Distributor ID: Ubuntu
Description:    Ubuntu 14.04.5 LTS
Release:        14.04
Codename:       trusty
```
Failures:
```bash
...
Err https://apt.dockerproject.org ubuntu-trusty/main amd64 Packages
  gnutls_handshake() failed: Handshake failed
...
W: Failed to fetch https://apt.dockerproject.org/repo/dists/ubuntu-trusty/main/binary-amd64/Packages  gnutls_handshake() failed: Handshake failed
```

**Comment by shrmrf** [comment]
btw, for all those having a problem, if I use `http://apt.dockerproject.org/repo ubuntu-trusty main` (no `HTTPS`)

**Comment by mlaventure** [comment]
@shrmrf could you do a `dig apt.dockerproject.org` please?

**Comment by pureclouds** [comment]
curl -sSL https://get.docker.com | sh fails on APT issue now. Quickest workaround for me now is:

curl -sSL https://get.docker.com | sed 's/https:\/\/apt.dockerproject.org/http:\/\/apt.dockerproject.org/' | sh

**Comment by thaJeztah** [comment]
@pureclouds the team is already looking into this issue, and it is narrowed down to an issue with some CDN mirrors (so depending on what mirror you end up with, the error occurs); can you give us the IP address that apt.dockerproject.org resolves to for you?

**Comment by cleiter** [comment]
@thaJeztah For me it's server-52-85-196-127.dub2.r.cloudfront.net (52.85.196.127)

**Comment by shrmrf** [comment]
@thaJeztah @mlaventure ... I'll do the `dig` but crazily, I'm also having similar problems with nodesource. Don't know if you share mirrors ( https://github.com/nodesource/distributions/issues/388 )

**Comment by pureclouds** [comment]
@thaJeztah 

~$ host apt.dockerproject.org
apt.dockerproject.org has address 52.85.196.127

**Comment by shrmrf** [comment]
```bash
$ dig apt.dockerproject.org

; <<>> DiG 9.10.3-P4-Ubuntu <<>> apt.dockerproject.org
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 15351
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 512
;; QUESTION SECTION:
;apt.dockerproject.org.		IN	A

;; ANSWER SECTION:
apt.dockerproject.org.	59	IN	A	52.222.171.192

;; Query time: 43 msec
;; SERVER: 127.0.1.1#53(127.0.1.1)
;; WHEN: Wed Nov 16 11:36:18 CET 2016
;; MSG SIZE  rcvd: 66

```

**Comment by thaJeztah** [comment]
yes, we identified `52.222.171.192` to be having the issue (`54.230.142.136` works)

**Comment by mlaventure** [comment]
I'd say to force using the node from @thaJeztah via `/etc/hosts` while we look at the issue

**Comment by WheresWardy** [comment]
@thaJeztah we're seeing the apt.dockerproject.org issue from
`server-52-85-68-116.lhr5.r.cloudfront.net. (52.85.68.116)`

but we're also seeing the same issue with get.docker.io (for old installs of lxc-docker) which appears to point directly to AWS but also seems to be having the problem?
```
$ host get.docker.io
get.docker.io is an alias for elb-io.us-east-1.aws.dckr.io.
elb-io.us-east-1.aws.dckr.io is an alias for us-east-1-elbio-rm5bon1qaeo4-623296237.us-east-1.elb.amazonaws.com.
us-east-1-elbio-rm5bon1qaeo4-623296237.us-east-1.elb.amazonaws.com has address 52.207.178.113
us-east-1-elbio-rm5bon1qaeo4-623296237.us-east-1.elb.amazonaws.com has address 54.85.12.131
us-east-1-elbio-rm5bon1qaeo4-623296237.us-east-1.elb.amazonaws.com has address 52.73.159.23
```

**Comment by mlaventure** [comment]
AFAICT this seems to be an issue with older version of gnutls only.

trusty packages are still based on gnutls26, xenial is now using gnutls30

**Comment by mokele** [comment]
I have been seeing this again today also, the work arounds above only get so far and am blocked at

`GPG error: https://apt.dockerproject.org ubuntu-trusty InRelease: The following signatures couldn't be verified because the public key is not available`

and am prompted that docker-engine is unauthenticated and am reluctant to force.

**Comment by DanielSWolf** [comment]
Just to let you know: My Linux-based Travis builds are now failing:

> W: Failed to fetch https://apt.dockerproject.org/repo/dists/ubuntu-trusty/main/binary-amd64/Packages  gnutls_handshake() failed: Handshake failed
> E: Some index files failed to download. They have been ignored, or old ones used instead.

**Comment by yermulnik** [comment]
Same here:
```
#> cat /etc/apt/sources.list.d/docker.list
deb https://get.docker.com/ubuntu docker main
#> apt-get update
[...]
W: Failed to fetch https://get.docker.com/ubuntu/dists/docker/main/binary-amd64/Packages  gnutls_handshake() failed: Handshake failed

W: Failed to fetch https://get.docker.com/ubuntu/dists/docker/main/binary-i386/Packages  gnutls_handshake() failed: Handshake failed

E: Some index files failed to download. They have been ignored, or old ones used instead.
#> dig +short get.docker.com
d3vuzgspmaykda.cloudfront.net.
52.222.157.234
52.222.157.250
52.222.157.40
52.222.157.189
52.222.157.94
52.222.157.60
52.222.157.188
52.222.157.43
```

**Comment by tiborvass** [comment]
FYI the /etc/hosts workaround worked for me. (Putting `54.230.142.136 apt.dockerproject.org` in /etc/hosts)

**Comment by thaJeztah** [comment]
A quick status update since my last comment; the infra team is now working on this issue as well, and is currently in contact with AWS. So far it looks like the issue is not on our side, but I'll post more information when it becomes available

**Comment by thaJeztah** [comment]
The team is still working together with AWS locating the problem.

**Comment by rianquinn** [comment]
Any idea on how to get Travis CI working again while we wait for a solution? All of our [travis ci](https://travis-ci.org/Bareflank/hypervisor) builds are down for this [repo](https://github.com/Bareflank/hypervisor)

**Comment by BanzaiMan** [comment]
@rianquinn we have a separate issue here: https://github.com/travis-ci/travis-ci/issues/6881#issuecomment-260948183

**Comment by thaJeztah** [comment]
AWS identified a possible cause and is rolling out updates. It may take some time for those to be available everywhere

**Comment by ayhid** [comment]
the problem seems to be fixed

**Comment by balakirevs** [comment]
true, for me it fixed

**Comment by yermulnik** [comment]
for it's fixed too

**Comment by ProgramItUp** [comment]
Interesting that you say it is fixed, I'm getting:

```
Err https://apt.dockerproject.org ubuntu-trusty/main amd64 Packages
  gnutls_handshake() failed: Decrypt error
Fetched 4,883 kB in 2min 4s (39.3 kB/s)
W: Failed to fetch https://apt.dockerproject.org/repo/dists/ubuntu-trusty/main/binary-amd64/Packages  gnutls_handshake() failed: Decrypt error

E: Some index files failed to download. They have been ignored, or old ones used instead.
```

**Comment by thaJeztah** [comment]
@ProgramItUp can you tell which IP address apt.dockerproject.org resolves to for you?

**Comment by ProgramItUp** [comment]
I'm on Ubuntu 14.04, on a machine behind a proxy.  So ping, traceroute, whois do not resolve.

What command would you like me to run?

**Comment by thaJeztah** [comment]
@ProgramItUp I just tried to reproduce on a fresh Ubuntu 14.04 droplet on DigitalOcean (AMS3), but cannot reproduce. The issue here (as you can see from the discussion above) was due to a configuration change in AWS cloudfront, and was only happening on some mirrors (IP-addresses).

If we know which IP-address your computer tries to connect to when accessing apt.dockerproject.org we would be able to verify if there is an issue again with their configuration, or if it's an issue in you local network setup (or proxy).

What does `host apt.dockerproject.org` show?

**Comment by thaJeztah** [comment]
It looks like something in your network is preventing you to reach apt.dockerproject.org. I suggest to ask in your company what's wrong. Perhaps it's blocked on purpose

**Comment by mokele** [comment]
Now that this bug is fixed for docker apt, the remaining bug for virtualbox users is to downgrade to 5.0.26 or the most recent TestBuilds (where the bug is already fixed)... see here https://www.virtualbox.org/ticket/16084

**Comment by Burstaholic** [comment]
Still happening for me:

```
W: Failed to fetch https://apt.dockerproject.org/repo/dists/ubuntu-trusty/main/binary-amd64/Packages  gnutls_handshake() failed: A TLS packet with unexpected length was received.

~> host apt.dockerproject.org
apt.dockerproject.org has address 52.85.211.244
```

**Comment by thaJeztah** [comment]
@Burstaholic `A TLS packet with unexpected length was received.` looks like a different message; could it be you are using a proxy?

**Comment by thaJeztah** [comment]
@Burstaholic just tried to do an install on 14.04 using `52.85.211.244` as IP address, and it worked for me, so looks like something in your setup (or proxy)

**Comment by Burstaholic** [comment]
You're right, I just noticed this is a different message, matching https://github.com/docker/docker/issues/27602 better, so I may need to talk to my networking guys.

**Comment by mokele** [comment]
@Burstaholic I suggest checking your VirtualBox version and downgrading or upgrade to TestBuild accordingly as mentioned above if >= 5.0.28, this looks like the same issue I had.

**Comment by Burstaholic** [comment]
This is actually a VM in a Hyper-V cluster, so it's probably not directly related.

**Comment by ProgramItUp** [comment]
Anybody know what the root cause of this is?   Looks like this problem has not gone away.

```
sudo apt-get update
....sucessful updates ....
Err https://apt.dockerproject.org ubuntu-trusty/main amd64 Packages
  gnutls_handshake() failed: Decrypt error
Err https://apt.dockerproject.org ubuntu-trusty/main i386 Packages
  gnutls_handshake() failed: Decrypt error
Ign https://apt.dockerproject.org ubuntu-trusty/main Translation-en_US
Ign https://apt.dockerproject.org ubuntu-trusty/main Translation-en
Fetched 3,343 kB in 2min 4s (26.8 kB/s)
W: Failed to fetch https://apt.dockerproject.org/repo/dists/ubuntu-trusty/main/binary-amd64/Packages  gnutls_handshake() failed: Decrypt error

W: Failed to fetch https://apt.dockerproject.org/repo/dists/ubuntu-trusty/main/binary-i386/Packages  gnutls_handshake() failed: Decrypt error
```

**Comment by mlaventure** [comment]
@ProgramItUp Are you behind a proxy? If so, that's usually a scenario where gnutls has issue. You may want to switch to openssl instead

**Comment by petewilcock** [comment]
I can also report this issue still occurring for me using https.

**Comment by fperret** [comment]
Same issue here for both http and https.

**Comment by thaJeztah** [comment]
@petewilcock @fperret if you followed the manual instructions on https://docs.docker.com/engine/installation/linux/ubuntu/ today, it could be a different issue; there's an error in the installation instructions, and because of that, an incorrect path is used to get the packages.

There's a number of pull requests open to fix this, and will be addressed later today; you can find the correct instructions in this PR; https://github.com/docker/docker.github.io/pull/1243/files

**Comment by NeoPolus** [comment]
I was having this issue today, and like @thaJeztah said, the fix in https://github.com/docker/docker.github.io/pull/1243/files worked for me.

**Comment by mlaventure** [comment]
@RickMohr `debian-trusty` is not a proper distribution name.

It should be `ubuntu-trusty` if you're on ubuntu or one of the debian variant otherwise (e.g. `debian-stretch`, `debian-jessie`)

Update your `/etc/apt/source.lilst.d/docker` accordingly.

**Comment by RickMohr** [comment]
Thanks. I also had to delete the offending repo from ` /etc/apt/sources.list`

**Comment by sunshineo** [comment]
I'm still having this issue.

**Comment by mlaventure** [comment]
@sunshineo if you don't provide any details, you can't expect us to help :sweat_smile:

**Comment by randy-jr-olive** [comment]
This is the error I get:

```
Err:3 https://apt.dockerproject.org/repo ubuntu-xenial/main amd64 Packages
  Received HTTP code 403 from proxy after CONNECT
```

And the following warnings and errors:
```
W: The repository 'https://apt.dockerproject.org/repo ubuntu-xenial Release' does not have a Release file.
N: Data from such a repository can't be authenticated and is therefore potentially dangerous to use.
N: See apt-secure(8) manpage for repository creation and user configuration details.
E: Failed to fetch https://apt.dockerproject.org/repo/dists/ubuntu-xenial/main/binary-amd64/Packages  Received HTTP code 403 from proxy after CONNECT
E: Some index files failed to download. They have been ignored, or old ones used instead.
```

my apt/sources.list entry is:
deb https://apt.dockerproject.org/repo/ ubuntu-xenial main

I am using Ubuntu server 16.04. I followed the install instructions at this page:
[https://docs.docker.com/engine/installation/linux/ubuntu/](https://docs.docker.com/engine/installation/linux/ubuntu/)

**Comment by thaJeztah** [comment]
I'm not able to reproduce that failure; "Received HTTP code 403 from proxy" are you behind a proxy server?

**Comment by randy-jr-olive** [comment]
Yes, I am! Good call, I use an apt-cacher proxy. I disabled it (the proxy) for this server in question, and it is working correctly now. I have successfully installed Docker. Is there something I could try doing to the proxy to get it to work?

**Comment by sunshineo** [comment]
@mlaventure I meant to just report the status. I already followed the work around by change from https to http and it worked. I just meant to tell owner that this is still happening for https. Don't think that this is fixed and close the issue since no one report it anymore.

**Comment by mlaventure** [comment]
@sunshineo if you have the gnutls issue it means you're on gnutls26, try updating to gnutls28

**Comment by sunshineo** [comment]
@mlaventure I'll check and report back.

**Comment by ChipmunkV** [comment]
> I'm not able to reproduce that failure; "Received HTTP code 403 from proxy" are you behind a proxy server?

Why it's so special that it doesn't work with a proxy server?

**Comment by thaJeztah** [comment]
> Why it's so special that it doesn't work with a proxy server?

Doesn't have to be "special", but can be the result of a misconfigured proxy or networking issue. That's not a bug that can be fixed here.

**Comment by MichaelDBA** [comment]
I am having this problem too still regardless of using http vs https
Err https://download.docker.com Trusty/stable amd64 Packages
  HttpError404
Ign https://download.docker.com Trusty/stable Translation-en_US
Ign https://download.docker.com Trusty/stable Translation-en
Fetched 48.7 kB in 3s (14.8 kB/s)
W: GPG error: https://apt.dockerproject.org ubuntu-trusty InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY F76221572C52609D
W: Failed to fetch https://download.docker.com/linux/ubuntu/dists/Trusty/stable/binary-amd64/Packages  HttpError404

E: Some index files failed to download. They have been ignored, or old ones used instead.

**Comment by thaJeztah** [comment]
@MichaelDBA looks like you still have the old `apt.dockerproject.org` repository configured as well; 

if there's a file named `/etc/apt/sources.list.d/docker.list` on your computer, remove it. Also check the content of `/etc/apt/sources.list` and check if the download.docker.com repository is configured properly; it should look something like:

```bash
$ cat /etc/apt/sources.list | grep docker

deb [arch=amd64] https://download.docker.com/linux/ubuntu trusty edge stable
# deb-src [arch=amd64] https://download.docker.com/linux/ubuntu trusty edge stable
```

**Comment by MichaelDBA** [comment]
Yes I finally got around my errors by editing sources.list file and changing Trusty to trusty (dont know how that got there).  Running the following to get rid of PUBLIC KEY NOT FOUND errors:
`apt-key adv --keyserver keyserver.ubuntu.com --recv-keys F76221572C52609D`

Manually created the  /etc/apt/sources.list.d/docker.list file and added this:
deb https://apt.dockerproject.org/repo ubuntu-trusty main

Then when i did an apt-cache search docker, the docker-ce package was now available

**Comment by thaJeztah** [comment]
You should no longer use the ` https://apt.dockerproject.org/repo` repository as it only contains packages that are EOL by now (latest version that was published there is docker 17.05)

**Comment by hansyson** [comment]
not working for me either
W: Failed to fetch https://download.docker.com/linux/ubuntu/dists/precise/stable/source/Sources

W: Failed to fetch https://download.docker.com/linux/ubuntu/dists/precise/stable/binary-amd64/Packages

E: Some index files failed to download. They have been ignored, or old ones used instead.

**Comment by thaJeztah** [comment]
@hansyson Ubuntu 12 hasn't been supported for a long time, and there's no Docker packages for that version. Ubuntu 12.04 also reached EOL last year so doesn't receive updates (including security patches) anymore; https://www.ubuntu.com/info/release-end-of-life

From the [installation documentation](https://docs.docker.com/install/linux/docker-ce/ubuntu/#os-requirements) of the current release;

> ## OS requirements
> 
> To install Docker CE, you need the 64-bit version of one of these Ubuntu versions:
>
> - Artful 17.10 (Docker CE 17.11 Edge and higher only)
> - Zesty 17.04
> - Xenial 16.04 (LTS)
> - Trusty 14.04 (LTS)

**Comment by czerasz** [comment]
I also run into the issue while updating the repos:

``` 
$ apt-get update
...
Ign http://downloads.linux.hpe.com trusty/current/non-free Translation-en 
Fetched 6,408 kB in 8s (728 kB/s) 
W: Failed to fetch https://download.docker.com/linux/ubuntu/dists/trusty/InRelease Unable to find expected entry 'main/binary-amd64/Packages' in Release file (Wrong sources.list entry or malformed file)

E: Some index files failed to download. They have been ignored, or old ones used instead. 
```

This was because the source was defined in multiple files:

``` 
$ cat /etc/apt/sources.list | grep docker 
deb [arch=amd64] https://download.docker.com/linux/ubuntu trusty stable 
# deb-src [arch=amd64] https://download.docker.com/linux/ubuntu trusty stable 
$ cat /etc/apt/sources.list.d/docker.list.save 
deb https://download.docker.com/linux/ubuntu trusty main 
$ cat /etc/apt/sources.list.d/docker.list 
deb https://download.docker.com/linux/ubuntu trusty main 
```

I removed `/etc/apt/sources.list.d/docker.list.save`, `/etc/apt/sources.list.d/docker.list` and removed the docker entry from `/etc/apt/sources.list`.

Next I followed the instructions from the [Dockers install documentation](https://docs.docker.com/install/linux/docker-ce/ubuntu/):

``` 
$ apt-get install -y \ 
apt-transport-https \ 
ca-certificates \ 
curl \ 
software-properties-common 
$ curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add - 
$ apt-key fingerprint 0EBFCD88 
$ add-apt-repository \ 
"deb [arch=amd64] https://download.docker.com/linux/ubuntu \ 
$(lsb_release -cs) \ 
stable" 
```

After the steps above I got a clean `apt-get update`.
